src/elements/basic/ImageElement.tsx

```diff
@@ -19,30 +19,39 @@ function ImageElement({
   elementProps = {},
   children
 }: any) {
-  const [imageUrl, setImageUrl] = useState(element.properties.source_image);
+  const [documentUrl, setDocumentUrl] = useState(
+    element.properties.source_image
+  );
+  const [documentType, setDocumentType] = useState<string | undefined>('');
   const [applyWidth, setApplyWidth] = useState(true);
-
   const styles = useMemo(
     () => applyImageStyles(element, responsiveStyles),
     [responsiveStyles]
   );
 
-  const imageField = element.properties.uploaded_image_file_field_key ?? '';
-  let imageFieldSource = fieldValues[imageField];
+  const fieldKey = element.properties.uploaded_image_file_field_key ?? '';
+  let imageFieldSource = fieldValues[fieldKey];
   if (imageFieldSource && Array.isArray(imageFieldSource))
     imageFieldSource = imageFieldSource[0];
-
   useEffect(() => {
     if (imageFieldSource) {
-      if (typeof imageFieldSource === 'string') setImageUrl(imageFieldSource);
-      else {
-        getThumbnailData(imageFieldSource).then((data) =>
-          setImageUrl(data.thumbnail)
-        );
+      if (typeof imageFieldSource === 'string') {
+        setDocumentType('');
+        setDocumentUrl(imageFieldSource);
+      } else {
+        getRenderData(imageFieldSource).then((data) => {
+          setDocumentType(data.type);
+          setDocumentUrl(data.url);
+        });
       }
-    } else setImageUrl(element.properties.source_image);
+    } else {
+      setDocumentUrl(element.properties.source_image);
+      setDocumentType('');
+    }
   }, [imageFieldSource, element.properties.source_image]);
 
+  const displayPDF = documentUrl && documentType === 'application/pdf';
+  const displayImage = !fieldKey || (documentUrl && !displayPDF);
```

Comment:
is this needed? since we can just do !displayPDF. If the image / field doesn't exist, our existing behavior is we still have the image element render a placeholder image